### PR TITLE
fix(edit-page-2): Fixed contentlets inserts in wrong position when using contentlet tools

### LINKED ISSUE #27650: Edit Page V2: Contentlet inserts in wrong position when using contentlet tools
### Parent Issue

https://github.com/dotCMS/core/issues/27546

### Problem Statement

When using contentlet tools (Blue plus button that appears on top and down of a contentlet), we can expect that any content type added with these buttons can be inserted in a wrong position. This can be a really weird behavior for the end user, because the contentlet will appear anywhere but the position the user selected.

### Steps to Reproduce

1. Open new Edit Page
2. Hover in a contentlet
3. Click in one of the blue plus buttons that appeared (A menu should appear)
4. Hover on another contentlet, make sure that the blue border and buttons appeared on the contentlet you hovered
5. Try to click on an option in the menu without hovering in another contentlet
6. Select a contentlet that is not duplicated
7. See how the contentlet was added in a different place that you selected in the 3rd step

## Screenshot following the steps and adding another cases

https://github.com/dotCMS/core/assets/63567962/979d725b-3135-44fe-9b83-e45930b4bd43

### Acceptance Criteria

* User should be able to add a contentlet in a selected position without problems

### dotCMS Version

`master-latest`

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### Insights

We have a problem with the flow of how this works.

This flow is not a staged process but a parallel one. This means that we can reset the contentlet we first selected as position, while still choosing which type of content are we going to add. This ends in this weird behavior, where we didn't selected any type of content but can still make hover in another contentlet and see the buttons and also click on them, resulting in a contentlet added in a wrong position if we make the correct steps.

In technical terms, when we hover on a contentlet the client sends a post message that tells the editor to set the new contentlet as the pivot for the position of the new contentlet (works as reference to add before or after of it ). And this post message is send even if we already selected a contentlet as pivot to add the new contentlet (click on the blue button).

Here are some Excalidraws to see the current flow and how it could be.

#### Current flow
https://excalidraw.com/#json=09ubKwCp97QNmphkp7gHA,8W1DTltNK4kQemabPXuvGg

#### Proposed Flow
https://excalidraw.com/#json=JXO8t9LJJGBEj469sp_xd,HnrijBbBI1E-UX5ubpHo0Q

### Assumptions & Initiation Needs

* We understand the event flow of Edit Page V2
* We understand the impact in a UX point of view

### Quality Assurance Notes & Workarounds

I would recommend to QA, that we need to test all the behaviors in all contentlets and containers, empty or not. We found this bug while testing the behavior of the add using contentlets that the client didn't have components for (No component for XYZ).

### PR DESCRIPTION
**Pull request by KevinDavilaDotCMS**
### Proposed Changes
* When the hover is changed to other contentlet, the menus are closed.
* If the user continue hovering the same contentlet (moving the mouse inside), nothing happens

With this, when the user hover other contentlet, the menus are closed and the reference to "old contentlet" is broken, so the user always have the right reference to insert other contentlet

https://github.com/dotCMS/core/assets/144152756/ed9058ea-a95d-4724-a60b-31aaee1b0b5e

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts [hunk 1]
```diff
@@ -72,6 +72,23 @@ describe('EmaContentletToolsComponent', () => {
             expect(spectator.query(byTestId('drag-image'))).toHaveText('test');
         });
 
+        it('should close menus when contentlet @input was changed', () => {
+            const spyHideMenus = jest.spyOn(spectator.component, 'hideMenus');
```
Comment (hmoreras) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts:76]: check for the menu's `hide()` instead ?

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts (hunk 1 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts:76]: We can only check the hide of one menu (If one is open and another is selected, the previous one is closed, so the .hide would be executed to undefined)

But I just added a case for one menu

Comment (fmontes) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.spec.ts:76]: Never mock a method on the component you're testing, mock the `menu.hide` and the `menuVtl.hide`

**Review by fmontes — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]
